### PR TITLE
refactor: node log

### PR DESCRIPTION
**Pull request by kobayu858**
## Description

- Corresponding logs to preemptive.
- However, the method of logging core run times has not been changed
　- To be changed in the next PR
## Related links

None.

## Pre-review checklist for the PR author

- [x] I've confirmed the [self-checklist](https://docs.google.com/spreadsheets/d/1p5KMcm752iCySHJz0xL1qf6fcLDKprbo/edit#gid=863618628).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: lib/src/scheduler.rs [hunk 1]
```diff
@@ -266,16 +268,15 @@ pub trait DAGSetSchedulerBase<T: ProcessorBase + Clone> {
         ready_nodes
     }
 
-    fn allocate_node(&mut self, node: &NodeData, core_i: usize) {
+    fn allocate_node(&mut self, node: &NodeData, core_i: usize, release_count: i32) {
```
Comment (atsushi421) [inline lib/src/scheduler.rs:271]:
```suggestion
    fn allocate_node(&mut self, node: &NodeData, core_i: usize, job_i: usize) {
```

File: lib/src/log.rs [hunk 2]
```diff
@@ -285,7 +291,7 @@ pub struct DAGSetSchedulerLog {
     dag_set_info: DAGSetInfo,
     processor_info: ProcessorInfo,
     dag_set_log: Vec<DAGLog>,
-    node_set_logs: Vec<Vec<NodeLog>>,
+    node_set_logs: Vec<Vec<JobLog>>, // node_set_logs[dag_id][job_index]
```
Comment (atsushi421) [inline lib/src/log.rs:294]: This comment is not valid with preemption.

**Review by atsushi421 — CHANGES_REQUESTED** [review]

File: lib/src/log.rs [hunk 3]
```diff
@@ -138,23 +129,38 @@ impl DAGLog {
     }
 }
 
-#[derive(Clone, Default, Serialize, Deserialize)]
-pub struct NodeLog {
-    core_id: Vec<usize>,
+#[derive(Clone, Serialize, Deserialize)]
+pub enum JobEvent {
```
[the remark this answers is not in the document]
Comment (kobayu858) [reply lib/src/log.rs:133]: Changed.

File: lib/src/scheduler.rs (hunk 1 above)
Comment (kobayu858) [reply lib/src/scheduler.rs]: Changed.

File: lib/src/log.rs (hunk 2 above)
Comment (kobayu858) [reply lib/src/log.rs]: Deleted.

**Review by atsushi421 — APPROVED** [review]
LTGM

**Pull request merged**